### DISCUSSION TITLE
로컬 캐싱

### DISCUSSION THREAD

**Discussion by jaeyeong815** [WantedPreonboardingFE5team/pre-onboarding-7th-2-1-5#42] [General]
과제 구현 목표에 보면 `API 호출별로 로컬 캐싱 구현`이 작성되어있는데

캐싱에 대해 검색해봤을 때 `이미 가져온 데이터나 계산된 결과값의 복사본을 저장함으로써 처리 속도를 향상시키며, 이를 통해 향후 요청을 더 빠르게 처리할 수 있다.` 이렇게 설명하고 있는데, 데이터를 로컬 스토리지에 저장하거나 indexDB를 사용하기도 하더라고요..?
근데 어떤 데이터를 저장해야하는지 잘 모르겠어서요

전체 데이터를 로컬에 저장하고 검색할 때마다 저장된 전체 데이터에서 뽑아 쓰는걸 의미하는 걸까요?
아니면 검색할 때마다 반환되는 데이터를 로컬에 저장하는걸 의미하는 걸까요?

제가 잘 이해를 못하고 있는 것 같아 한번 여쭤봅니당,,

**Comment by hyoniiii** [comment]
검색 할 때마다 반환되는 결과를 로컬에 저장하여 캐싱한 후, 중복 결과 값이 있을 경우 api를 호출하지 않고 캐시를 사용하고
결과 값이 없을 경우 api를 호출하게끔 해서 api 최적화를 시키라는 걸로 저는 이해했어요! 

미흡한 설명이지만 이해가 되셨으면 좋겠습니다🥲

**Reply by jaeyeong815** [reply]
아하 이해했습니다! 검색 시 반환되는 데이터군여! 감사합니다 ~,~
